### PR TITLE
Introduce PHPStan at level 1

### LINKED ISSUE #656: Static analysis
I had a great experience with static analysis tools such as https://github.com/phpstan/phpstan

Do you think we should use it for Lighthouse too? Are there other tools that work better?

### PR DESCRIPTION
**Pull request by spawnia**
Resolves #656

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -51,6 +52,7 @@ install:
 
 script:
   - if [[ $COVERAGE = 1 ]]; then phpdbg -qrr vendor/bin/phpunit --coverage-clover=coverage.xml; fi
+  - if [[ $STATIC_ANALYSIS = 1 ]]; then composer require nunomaduro/larastan && composer stan; fi
```
Comment (lucasmichot) [inline .travis.yml]: So this means the check will only be done by Travis

File: composer.json [hunk 2]
```diff
@@ -38,6 +38,7 @@
         "mockery/mockery": "^1.0",
         "orchestra/database": "3.5.*|3.6.*|3.7.*|3.8.x-dev",
         "orchestra/testbench": "3.5.*|3.6.*|3.7.*|3.8.*",
+        "phpstan/phpstan": "^0.11.5",
```
Comment (lucasmichot) [inline composer.json]: In this case, should this required in travis file only?

**Comment by spawnia** [comment]
@lucasmichot the issue i am working around is the incompatibility of Larastan with Laravel 5.5

File: composer.json (hunk 2 above)
Comment (spawnia) [reply composer.json]: Done

File: .travis.yml (hunk 1 above)
Comment (spawnia) [reply .travis.yml]: At the moment, yes.

**Pull request merged**